### PR TITLE
go get -t -u ./... && go mod tidy

### PR DESCRIPTION
**Pull request by bdunne**
- Upgrade to sigs.k8s.io/controller-runtime@v0.16.2

### CODE REVIEW

File: manageiq-operator/cmd/main.go [hunk 1]
```diff
@@ -30,9 +30,11 @@ import (
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlcache "sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics/server"
```
Comment (Fryguy) [inline manageiq-operator/cmd/main.go:37]: Not that it really matter, but I would expect this to be next to the other new one sort of alphabetically?

File: manageiq-operator/go.mod [hunk 2]
```diff
@@ -44,34 +44,30 @@ require (
...
+	k8s.io/kube-openapi v0.0.0-20230918164632-68afd615200d // indirect
+	k8s.io/utils v0.0.0-20230726121419-3b25d923346b // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.3.0 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
-
-replace github.com/gogo/protobuf v1.3.1 => github.com/gogo/protobuf v1.3.2 // Fix CVE-2021-3121
-
-replace k8s.io/client-go v0.27.4 => k8s.io/client-go v0.28.0-rc.0 // https://github.com/kubernetes/client-go/issues/1265
```
Comment (Fryguy) [inline manageiq-operator/go.mod:77]: NICE!  😄

**Review by Fryguy — APPROVED** [review]

File: manageiq-operator/cmd/main.go (hunk 1 above)
Comment (bdunne) [reply manageiq-operator/cmd/main.go:37]: Yeah, me too. Unfortunately the tooling keeps moving it down there 🤦🏻‍♂️

**Comment by Fryguy** [comment]
Backported to `quinteros` in commit 421c69cfa2c9d755838d17fe11846c2fdbf3e2a3.

```text
commit 421c69cfa2c9d755838d17fe11846c2fdbf3e2a3
Author: Jason Frey <fryguy9@gmail.com>
Date:   Tue Sep 26 14:04:47 2023 -0400

    Merge pull request #994 from bdunne/Update_dependencies
    
    go get -t -u ./... && go mod tidy
    
    (cherry picked from commit 8be17ba6809ebca01deff7d6ec9f6a97ce8d8f9e)
```

**Pull request merged**